pdal/util/ProgramArgs.hpp

```diff
@@ -516,11 +547,12 @@ class TArg<bool> : public Arg
       \param description  Argument description.
       \param variable  bool variable to which the value of the argument should
         be bound.
+      \param required  Is field required or not.
     */
     TArg(const std::string& longname, const std::string& shortname,
-        const std::string& description, bool& variable) :
-        Arg(longname, shortname, description), m_val(variable),
-        m_defaultVal(false), m_defaultProvided(false)
+         const std::string& description, bool& variable, const bool required) :
+         Arg(longname, shortname, description, required, TypeName<bool>::get()), m_val(variable),
+         m_defaultVal(false), m_defaultProvided(false)
```

Comment:
It doesn't look like a good idea to change every constructor, but if it's a good approach than ok.